apis/google/gmail/code/updateDraft.py

```diff
@@ -0,0 +1,73 @@
+import base64
+import os
+from email.mime.text import MIMEText
+
+from googleapiclient.errors import HttpError
+
+from auth import client
+
+
+def main():
+    to_emails = os.getenv('TO_EMAILS')
+    if to_emails is None:
+        raise ValueError("At least one recipient must be specified with 'to_emails'")
+
+    cc_emails = os.getenv('CC_EMAILS')
+    bcc_emails = os.getenv('BCC_EMAILS')
+    subject = os.getenv('SUBJECT')
+    if subject is None:
+        raise ValueError("Email subject must be set")
+
+    message = os.getenv('MESSAGE')
+    if message is None:
+        raise ValueError("Email message must be set")
+
+    draft_id = os.getenv('DRAFT_ID')
+    if draft_id is None:
+        raise ValueError("draft_id must be set")
+
+    service = client('gmail', 'v1')
+    try:
+        update_draft(service=service, draft_id=draft_id, to=to_emails, cc=cc_emails, bcc=bcc_emails, subject=subject,
+                     body=message)
+    except HttpError as err:
+        print(err)
+
+
+def get_draft(service, draft_id):
+    try:
+        draft = service.users().drafts().get(userId='me', id=draft_id).execute()
+        return draft
+    except HttpError as error:
+        print(f'An error occurred retrieving the draft: {error}')
+        return None
+
+
+def update_draft(service, draft_id, to, cc, bcc, subject, body):
+    try:
+        message = create_message(to=to, cc=cc, bcc=bcc, subject=subject, message_text=body)
+        updated_draft = {
+            'id': draft_id,
+            'message': message,
+        }
+
+        draft_response = service.users().drafts().update(userId='me', id=draft_id, body=updated_draft).execute()
+        print(f"Draft Id: {draft_response['id']} - Draft updated successfully!")
+    except HttpError as error:
+        print(f'An error occurred: {error}')
+
+
+def create_message(to, cc, bcc, subject, message_text):
+    message = MIMEText(message_text)
+    message['to'] = to
+    if cc is not None:
+        message['cc'] = cc
+    if bcc is not None:
+        message['bcc'] = bcc
+    message['subject'] = subject
+    raw_message = base64.urlsafe_b64encode(message.as_bytes()).decode('utf-8')
+    return {'raw': raw_message}
```

Comment:
I think this function has appeared two or three times - can this and other repeated code be put into another place that is imported each time it's needed?

Reply:
fixed